File: javascript/isbn-verifier/isbn-verifier.js
export const isValid = (number) => {
  let formattedNumber = [...number].filter((value, index) => {
    return value.match(/[0-9]/) || value === 'X' && index === number.length - 1;
  }).map(value => value === 'X' ? 10 : value);
  return formattedNumber.length !== 10 ? false : formattedNumber.reduce((sum, value, index) => sum + value * (10 - index), 0) % 11 === 0;
};

File: python/isbn-verifier/isbn_verifier.py
def is_valid(isbn):
    digits = [digit for digit in isbn if digit.isalnum()]
    if len(digits) != 10 or [digit for index, digit in enumerate(digits) if digit.isnumeric() or index == len(digits) - 1 and digit == 'X'] != digits:
        return False
    return sum([(int(value) if value != 'X' else 10) * (index + 1) for index, value in enumerate(digits)]) % 11 == 0
